.github/workflows/continuous-integration.yml

```diff
@@ -0,0 +1,99 @@
+
+name: "Continuous Integration"
+
+on:
+    push:
+        branches:
+            - master
+    pull_request:
+
+jobs:
+  phpunit:
+    name: "PHPUnit"
+    runs-on: "ubuntu-latest"
+    env:
+      SYMFONY_REQUIRE: ${{matrix.symfony-require}}
+      SYMFONY_DEPRECATIONS_HELPER: ${{matrix.symfony-deprecations-helper}}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        php-version:
+          - "7.3"
+          - "7.4"
+          - "8.0"
+        dependencies:
+          - "php-http/guzzle6-adapter"
+        stability:
+          - "stable"
+        symfony-require:
+          - ""
+        symfony-deprecations-helper:
+          - ""
+        composer-options:
+            - ""
+        include:
+          # Minimum supported Symfony version and lowest PHP version
+          - dependencies: "php-http/guzzle7-adapter"
+            php-version: "7.3"
+            composer-options: "--prefer-stable --prefer-lowest"
+            symfony-deprecations-helper: "weak"
+
+          # Test the latest stable release
+          - dependencies: "php-http/guzzle7-adapter php-http/vcr-plugin:^1.0@dev"
+            php-version: "7.3"
+          - php-version: "7.4"
+          - php-version: "8.0"
+
+          # Test with httplug 1.x clients
+          - dependencies: "php-http/buzz-adapter:^1.0 php-http/guzzle6-adapter:^1.1.1 php-http/react-adapter:^0.2.1 php-http/socket-client:^1.0"
+            php-version: "7.3"
+          # Test with httplug 2.x clients
+          - dependencies: "php-http/guzzle7-adapter php-http/curl-client:^2.0.0 php-http/vcr-plugin:^1.0@dev"
+            php-version: "7.3"
+
+          # Latest commit to master
+          - dependencies: "php-http/guzzle7-adapter php-http/vcr-plugin:^1.0@dev"
+            php-version: "7.0"
+            stability: "dev"
+          - php-version: "8.0"
+            stability: "dev"
+
+          # Test maintained versions of Symfony
+          - symfony-require: "3.4.*"
+            php-version: "7.3"
+          - dependencies: "php-http/guzzle7-adapter symfony/http-client:^4.4"
+            symfony-require: "4.4.*"
+            php-version: "7.3"
+          - dependencies: "php-http/guzzle7-adapter symfony/http-client:^5.0"
+            symfony-require: "5.0.*"
```

Comment:
i guess this comes from the previous config, but current would be 5.3.* ... 

but lets port the setup first, without introducing additional changes.